### ISSUE TITLE
 DateTime UnmarshalBSONValue panics on a null bson value

### ISSUE THREAD

**Issue by dimovnike** [go-openapi/strfmt#97]
```
// UnmarshalBSONValue is an interface implemented by types that can unmarshal a
// BSON value representation of themselves. The BSON bytes and type can be
// assumed to be valid. UnmarshalBSONValue must copy the BSON value bytes if it
// wishes to retain the data after returning.
func (t *DateTime) UnmarshalBSONValue(tpe bsontype.Type, data []byte) error {
	i64 := int64(binary.LittleEndian.Uint64(data))
	// TODO: Use bsonprim.DateTime.Time() method
	*t = DateTime(time.Unix(i64/1000, i64%1000*1000000))

	return nil
}
```

`binary.LittleEndian.Uint64()` panics when the len(data) is 0. 
What is the intended usage for this? how to handle bson null values?

**Comment by youyuanwu** [comment]
Maybe this Unmarshal function needs to be reworked based on https://pkg.go.dev/go.mongodb.org/mongo-driver/bson/primitive#DateTime

**Comment by dimovnike** [comment]
i closed the PR, i will get back about this.

**Comment by dimovnike** [comment]
here is a better fix: https://github.com/go-openapi/strfmt/pull/99 .

**Comment by fredbi** [comment]
Fix #99 has been merged.
